server/routes/teacher/test.js

```diff
@@ -1,3 +1,27 @@
 const route = require('express').Router();
+const generalApi = require('../../db-middleware/general-api');
+const testApi = require('../../db-middleware/test-api');
+
+route.get('/get-students/:teacherId', (req, res) => {
+  generalApi
+    .getFilteredStudentsByTeacher(req.params.teacherId)
+    .then((studId) => {
+      res.status(200).send(studId);
+    })
+    .catch((err) => {
+      res.status(404).send(err);
+    });
+});
+
+route.get('/pending-requests/:teacherId', (req, res) => {
+  testApi
+    .allPendingRequestsByMyPupils(req.params.teacherId)
```

Comment:
Method name should include action. And also ByMyPupils is noq quiet right, according to our naming style by should refer to the param.
Consider naming "getPendingrequestByTeacher"